src/planning/strategizer.ts

```diff
@@ -220,71 +223,29 @@ export class Strategizer {
     return mergedEvaluations;
   }
 
-  static over(results, walker: StrategizerWalker, strategy: Strategy): Descendant[] {
-    walker.onStrategy(strategy);
-    results.forEach(result => {
-      walker.onResult(result);
-      walker.onResultDone();
-    });
-    walker.onStrategyDone();
-    return walker.descendants;
-  }
 }
 
-export interface Descendant {result; score: number; derivation; hash; valid: boolean; errors?; normalized?;}
-
-export abstract class StrategizerWalker {
-  descendants: Descendant[];
-  currentStrategy;
-  currentResult;
-
-  protected constructor() {
-    this.descendants = [];
-  }
-
-  onStrategy(strategy: Strategy) {
-    this.currentStrategy = strategy;
-  }
-
-  onResult(result): void {
-    this.currentResult = result;
-  }
-
-  createDescendant(result, score, hash, valid): void {
-    assert(this.currentResult, 'no current result');
-    assert(this.currentStrategy, 'no current strategy');
-    if (this.currentResult.score) {
-      score += this.currentResult.score;
-    }
-    this.descendants.push({
-      result,
-      score,
-      derivation: [{parent: this.currentResult, strategy: this.currentStrategy}],
-      hash,
-      valid,
-    });
+export class StrategizerWalker extends RecipeWalker {
+  constructor(tactic) {
+    super(tactic);
   }
 
-  onResultDone(): void {
-    this.currentResult = undefined;
+  createDescendant(result, score,): void {
```

Comment:
What's up with the comma after _score_? I just learned they are valid, but do you want one here?

Reply:
Just a copy-and-paste typo. Thanks for spotting it.